lib/lte_link_control/lte_lc.c

```diff
@@ -931,6 +935,57 @@ int lte_lc_edrx_req(bool enable)
 	return err;
 }
 
+int lte_lc_rai_req(bool enable)
+{
+	int err;
+	enum lte_lc_system_mode mode;
+
+	err = lte_lc_system_mode_get(&mode);
+	if (err) {
+		return err;
+	}
+
+	switch (mode) {
+	case LTE_LC_SYSTEM_MODE_LTEM:
+	case LTE_LC_SYSTEM_MODE_LTEM_GPS:
+		LOG_ERR("RAI not supported for LTE-M networks");
+		return -EOPNOTSUPP;
+	case LTE_LC_SYSTEM_MODE_NBIOT:
+	case LTE_LC_SYSTEM_MODE_NBIOT_GPS:
+		break;
+	default:
+		LOG_ERR("Unknown system mode");
+		LOG_ERR("Cannot request RAI for unknown system mode");
+		return -EOPNOTSUPP;
+	}
+
+	if (enable) {
+		char rai_req[10];
+
+		snprintf(rai_req, sizeof(rai_req), "AT%%XRAI=%s", rai_param);
+		err = at_cmd_write(rai_req, NULL, 0, NULL);
+	} else {
+		err = at_cmd_write(rai_disable, NULL, 0, NULL);
+	}
+
+	return err;
+}
+
+int lte_lc_rai_param_set(const char *value)
+{
+	if (value == NULL || strlen(value) != 1) {
+		return -EINVAL;
+	}
+
+	if (value[0] == '3' || value[0] == '4') {
+		memcpy(rai_param, value, strlen(value) + 1);
```

Comment:
At this point we know the string is of length 1, so can just enter 2 here to save a few CPU cycles